### PR TITLE
Onboard module into ibm registry

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

Onboarding to IBM Registry

### Release required?

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: ibm_catalog.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      "product_kind": "module",
+      "tags": [
+        "dev_ops",
+        "target_terraform",
+        "terraform",
+        "module"
+      ],
+      "keywords": [
+        "terraform",
+        "key protect",
```
Comment (ocofaigh) [inline ibm_catalog.json:15]: can keywords have spaces? maybe they can?

File: ibm_catalog.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        "authorization",
```
Comment (ocofaigh) [inline ibm_catalog.json:16]: Perhaps encryption instead of authorization?

**Comment by rajatagarwal-ibm** [comment]
Adding "cra-config" in a separate PR first.

File: ibm_catalog.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          "description": "Configure a metrics policy for the Key Protect instance."
+        },
+        {
+          "title": "Configure key create and import access policy",
+          "description": "Configure a key create and import access policy for the Key Protect instance."
+        },
+        {
+          "title": "Attach access tags",
+          "description": "Attach access tags to the Key Protect instance."
+        }
```
Comment (ocofaigh) [inline ibm_catalog.json:47]: Perhaps also add "Attach resource tags"

**Review by ocofaigh — CHANGES_REQUESTED** [review]
some small comments

File: ibm_catalog.json (hunk 1 above)
Comment (rajatagarwal-ibm) [reply ibm_catalog.json:15]: I assume they can as many services have multi-word names

**Review by SirSpidey — DISMISSED** [review]
Committed some description updates. 

We still need the placeholder diagram, caption, and description, right?

**Comment by rajatagarwal-ibm** [comment]
@SirSpidey placeholder diagram for the example is mentioned in the ibm_catalog.json file along with some description. Are you looking for something else here?

**Comment by SirSpidey** [comment]
> @SirSpidey placeholder diagram for the example is mentioned in the ibm_catalog.json file along with some description. Are you looking for something else here?

Nope. Just missed it and forgot that the digram is not hosted in the repo.

File: ibm_catalog.json [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        "encryption",
+        "policy"
+      ],
+      "short_description": "Terraform module to create and work with IBM Key Protect",
+      "long_description": "Use this module to provision and configure an IBM Key Protect instance.\nYou can configure the following aspects of your instance:\n- [Key Protect instance](https://cloud.ibm.com/docs/key-protect?topic=key-protect-about)\n- [Rotation policy](https://cloud.ibm.com/docs/key-protect?topic=key-protect-set-rotation-policy&interface=ui)\n- [Dual authorization policy](https://cloud.ibm.com/docs/key-protect?topic=key-protect-manage-dual-auth)\n- [Metrics policy](https://cloud.ibm.com/docs/key-protect?topic=key-protect-manage-monitor-metrics)\n- [Key create and import access policy](https://cloud.ibm.com/docs/key-protect?topic=key-protect-manage-keyCreateImportAccess)\n#### Usage\n```hcl\nmodule \"key_protect_module\" {\n  source  = \"terraform-ibm-modules/key-protect/ibm\"\n  version = \"latest\" # Replace \"latest\" with a release version to lock into a specific release\n  key_protect_name  = \"my-key-protect-instance\"\n  resource_group_id = \"xxXXxxXXxXxXXXXxxXxxxXXXXxXXXXX\"\n  region            = \"us-south\"\n}```",
```
Comment (ocofaigh) [inline ibm_catalog.json:20]: - source needs to be updated to `https://cm.globalcatalog.cloud.ibm.com/api/v1-beta/offering/source?archive=tgz&kind=terraform&name=terraform-ibm-key-protect&version=X.X.X`
- Move this comment so it shows above the source: `# Replace "X.X.X" with a release version to lock into a specific release`
- Missing a `\n` at the end. The ``` should be on the last line:
image

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comment

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**